winui/DataGrid/Export-To-Excel.md

```diff
@@ -0,0 +1,653 @@
+    //values while exporting to excel.
+    if (e.CellType == ExportCellType.RecordCell && e.ColumnName == "IsShipped")
+    {
+
+        //if the cell value is True, "Y" will be displayed else "N" will be displayed.
+
+        if (e.CellValue.Equals(true))
+            e.Range.Cells[0].Value = "Y";
+
+        else
+            e.Range.Cells[0].Value = "N";
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-cell-value-customization.png" alt="Customizing Cell Values while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, cell values are changed for `Is Shipped` column based on custom condition.
+
+### Changing row style in excel based on data
+
+You can customize the rows based on the record values by using `CellsExportHandler`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.CellsExportHandler = CellsExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void CellsExportHandler(object sender, DataGridCellExcelExportOptions e)
+{
+    var record = e.NodeEntry;
+
+    if (record != null && (record as OrderInfo).Country == "Mexico")
+    {
+        e.Range.CellStyle.ColorIndex = ExcelKnownColors.Green;
+        e.Range.CellStyle.Font.Color = ExcelKnownColors.White;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-row-style-customization.png" alt="Customizing Row Style based on Data while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, records having the `Country` name as `Mexico` are customized.
+
+### Customize the cells based on Column Name
+
+You can customize the cells based on `DataGridCellExcelExportOptions.ColumnName` property in the `CellsExportHandler`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.CellsExportHandler = CellsExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void CellsExportHandler(object sender, DataGridCellExcelExportOptions e)
+{
+
+    if (e.ColumnName != "OrderID")
+        return;
+
+    e.Range.CellStyle.Font.Size = 12;
+    e.Range.CellStyle.Font.Color = ExcelKnownColors.Pink;
+    e.Range.CellStyle.Font.FontName = "Segoe UI";
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-export-cell-value-based-on-column.png" alt="Customizing Cell Values based on Column name while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, `OrderID` column cells are customized while exporting.
```

Comment:
Here, the `OrderID` column cells are customized while exporting.

Reply:
Content modified